### PR TITLE
[WIP] Add option to declare list of authors from which the tweets are allowed

### PR DESCRIPTION
**Pull request by jacek-foremski**
> [EZP-27294](https://jira.ez.no/browse/EZP-27294)
#
This PR aims at adding the possibility to declare a list of allowed tweet authors. 
This feature currently is signalled throughout the code and in the tutorial documentation but is not implemented fully. My PR adds the missing pieces.

**TODO:**
- [ ] Rebase to the new branches

### CODE REVIEW

File: Form/StringToArrayTransformer.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+
+namespace EzSystems\TweetFieldTypeBundle\Form;
+
+use Symfony\Component\Form\DataTransformerInterface;
+
+class StringToArrayTransformer implements DataTransformerInterface
```
Comment (bdunogier) [inline Form/StringToArrayTransformer.php]: I'd add phpdoc stating that it uses a comma separated format.

File: eZ/Publish/FieldType/Tweet/LegacyConverter.php [hunk 2]
```diff
@@ -30,12 +30,20 @@ public function toFieldValue(StorageFieldValue $value, FieldValue $fieldValue)
 
     public function toStorageFieldDefinition(FieldDefinition $fieldDef, StorageFieldDefinition $storageDef)
     {
-        // @todo implement me
+        $storageDef->dataText1 =
```
Comment (bdunogier) [inline eZ/Publish/FieldType/Tweet/LegacyConverter.php]: Formatting nitpick: I'd go for this:
```php
$storageDef->dataText1 = json_decode(
  $fieldDef->fieldTypeConstraints->validators['TweetValueValidator']['authorList']
);
```

**Review by bdunogier — COMMENTED** [review]
Looks okay to me.

@mere-teresa do you feel confident enough about this to do the redacting part on your own, or would you rather get some tech content from @natanael89 first ? In that case, it would as well be part of this PR...

**Comment by jacek-foremski** [comment]
@bdunogier I added your suggestions and I also reversed if-clauses in StringToArrayTransformer, because guard clauses should be at the top.

@mere-teresa Let me know what is needed, I'll be happy to help.

**Review by bdunogier — APPROVED** [review]

**Pull request closed without merging**